### PR TITLE
Revit UTC

### PR DESCRIPTION
**Pull request by aosyatnik**
### Purpose

Link to YouTrack:
[MAGN-8993](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-8993) Correct UTC offset for Solar Radiation workflow from Revit.

After some investigation I came up to the solution, that the problem is that it's used daylight saving time.
Look at those images:

image

image

But then I faced with the problem. Revit API returns just datetime. But there is no means to detect if this is the summer time. Moreover, you event don't know time zone!
Revit API returns next things: datetime, latitude, longitude and useless name of the city/region, like "Boston, MA" (have no idea, how we can use it... especially if it is not famous place).

I found one project, that can translate  latitude and longitude to timezone. 
Here is link: https://github.com/mj1856/GeoTimeZone
But it translates coordinates into Olson time zone. So I had to create one more additional method `OlsonTimeZoneToTimeZoneInfo`, where I convert Olson time zone into TimeZoneInfo.

For now I won't create 2015, 2017 branches. Maybe, it's wrong direction. And I chose bad solution.

## Results:

Now they are the same.

image

image

### Declarations

- [ ] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.

### Reviewers

@mjkkirschner 

### FYIs

@jnealb

### CODE REVIEW

**Comment by QilongTang** [comment]
Happen to stop by here in this PR. After reading http://web.cs.ucla.edu/~eggert/tz/tz-link.htm. Looks like if we want to do it locally, the absolute right way is to integrate the timezone database (http://www.iana.org/time-zones). Not sure how much effort will it be though. There are tons of webAPI for converting (e.g. Google), but I guess they are off the table. You can find lots of info in this article.

File: src/Libraries/RevitNodes/RevitNodes.csproj [hunk 1]
```diff
@@ -45,6 +45,9 @@
       <HintPath>$(DYNAMOAPI)\DynamoServices.dll</HintPath>
       <Private>False</Private>
     </Reference>
+    <Reference Include="GeoTimeZone">
+      <HintPath>..\..\packages\GeoTimeZone.1.1.0\lib\portable-net40+sl50+win+wp80\GeoTimeZone.dll</HintPath>
```
Comment (mjkkirschner) [inline src/Libraries/RevitNodes/RevitNodes.csproj]: if we are using this package, we should add license information to the repo:
https://raw.githubusercontent.com/mj1856/GeoTimeZone/master/LICENSE

**Comment by mjkkirschner** [comment]
another potential library to investigate: https://github.com/nodatime/nodatime

File: src/Libraries/RevitNodes/RevitNodes.csproj (hunk 1 above)
Comment (mjkkirschner) [inline src/Libraries/RevitNodes/RevitNodes.csproj]: is this all we need to do to get nuget to download the package?

File: src/Libraries/RevitNodes/Elements/SunSettings.cs [hunk 2]
```diff
@@ -143,8 +170,165 @@ internal static DateTime TranslateTime(DateTime utc)
...
+        { "Indian/Mauritius", "Mauritius Standard Time" },
+        { "Pacific/Apia", "Samoa Standard Time" },
+        { "Pacific/Auckland", "New Zealand Standard Time" },
+        { "Pacific/Fiji", "Fiji Standard Time" },
+        { "Pacific/Guadalcanal", "Central Pacific Standard Time" },
+        { "Pacific/Guam", "West Pacific Standard Time" },
+        { "Pacific/Honolulu", "Hawaiian Standard Time" },
+        { "Pacific/Pago_Pago", "UTC-11" },
+        { "Pacific/Port_Moresby", "West Pacific Standard Time" },
+        { "Pacific/Tongatapu", "Tonga Standard Time" }
```
Comment (mjkkirschner) [inline src/Libraries/RevitNodes/Elements/SunSettings.cs]: seems like maybe we are missing some mappings...http://unicode.org/repos/cldr-tmp/trunk/diff/supplemental/zone_tzid.html

File: src/Libraries/RevitNodes/Elements/SunSettings.cs [hunk 3]
```diff
@@ -143,8 +170,165 @@ internal static DateTime TranslateTime(DateTime utc)
...
+        {
+            var windowsTimeZoneId = default(string);
+            var windowsTimeZone = default(TimeZoneInfo);
+            if (olsonWindowsTimes.TryGetValue(olsonTimeZoneId, out windowsTimeZoneId))
+            {
+                try { windowsTimeZone = TimeZoneInfo.FindSystemTimeZoneById(windowsTimeZoneId); }
+                catch (TimeZoneNotFoundException) { }
+                catch (InvalidTimeZoneException) { }
+            }
+            return windowsTimeZone;
```
Comment (mjkkirschner) [inline src/Libraries/RevitNodes/Elements/SunSettings.cs]: if we can't find the id what is the result of returning null?

**Comment by mjkkirschner** [comment]
@aosyatnik I have a few questions on this solution - thanks for investigating these issues and coming up with this solution, like @QilongTang says, I think you are on the right track - 

in addition to this work, can we add a few tests?

File: src/Libraries/RevitNodes/RevitNodes.csproj (hunk 1 above)
Comment (aosyatnik) [inline src/Libraries/RevitNodes/RevitNodes.csproj]: I guess we need `packages.config` file. It looks like it's all, that we need...

**Pull request closed without merging**